jobsdb/tx.go

```diff
@@ -0,0 +1,106 @@
+package jobsdb
+
+import (
+	"database/sql"
+	"errors"
+	"sync"
+)
+
+// ErrEmptyxTx sentinel error indicating an Empty Tx
+var ErrEmptyxTx = errors.New("jobsdb: empty tx")
+
+// TxGetter provides access to an sql.Tx
+type TxGetter interface {
+	// Tx retrieves the tx or returns an error if the tx could not be retrieved (started)
+	Tx() (*sql.Tx, error)
+	// MustTx retrieves the tx or panics in case of an error
+	MustTx() *sql.Tx
```

Comment:
qq: do you know when `Tx()` would actually return an error?

Reply:
if a database driver error is returned